### PR TITLE
Demo playback

### PR DESCRIPTION
**Pull request by d3nwah**
Ready to merge

### CODE REVIEW

File: Source/AliveLib/Scrab.cpp [hunk 1]
```diff
@@ -1789,28 +1789,22 @@ __int16 Scrab::AI_Fighting_2_4A5840()
...
         }
         else
         {
-            if (Handle_SlamDoor_or_EnemyStopper_4A4830(field_C4_velx, 0) &&
-                Check_IsOnEndOfLine_408E90(0, 2) &&
-                !vIsObjNearby_4253B0(ScaleToGridSize_4498B0(field_CC_sprite_scale) * FP_FromInteger(10), pTarget))
+            xPos += FP_FromInteger(50);
+        }
+        if (!Handle_SlamDoor_or_EnemyStopper_4A4830(field_C4_velx, 0) &&
+            !Check_IsOnEndOfLine_408E90(1, 2) &&
```
Comment (paulsapps) [inline Source/AliveLib/Scrab.cpp]: Check_IsOnEndOfLine_408E90(0, 2) isn't done anymore ?

Comment (d3nwah) [reply Source/AliveLib/Scrab.cpp]: fixed

File: Source/AliveLib/Well.cpp [hunk 2]
```diff
@@ -158,6 +158,17 @@ void Well::vScreenChanged_4E3070()
     field_6_flags.Set(BaseGameObject::eDead);
 }
 
+__int16 Well_NextRandom()
```
Comment (paulsapps) [inline Source/AliveLib/Well.cpp]: mark static

File: Source/AliveLib/Well.cpp [hunk 3]
```diff
@@ -158,6 +158,17 @@ void Well::vScreenChanged_4E3070()
...
+{
+    sWellRndSeed_563AA0++;
+
+    if (sWellRndSeed_563AA0 > 255)
+    {
+        sWellRndSeed_563AA0 = 1;
```
Comment (paulsapps) [inline Source/AliveLib/Well.cpp]: didn't it reset to 0 not 1?

Comment (d3nwah) [reply Source/AliveLib/Well.cpp]: yeah, but it still needs to be incremented by 1.

**Pull request merged**